packages/contractkit/src/wrappers/DowntimeSlasher.ts

```diff
@@ -108,87 +195,89 @@ export class DowntimeSlasherWrapper extends BaseWrapper<DowntimeSlasher> {
   }
 
   /**
-   * Slash a Validator for downtime.
-   * @param validator Validator account or signer to slash for downtime.
-   * @param startBlock First block of the downtime, undefined if using endBlock.
-   * @param endBlock Last block of the downtime. Determined from startBlock or grandparent of latest block if not provided.
+   * Returns true if the validator did not sign any blocks for the specified overlapping or adjacent
+   * intervals.
+   * @param validatorOrSignerAddress Address of the validator account or signer.
+   * @param startBlocks A list of interval start blocks for which signature bitmaps have already
+   * been set.
+   * @param endBlocks A list of interval end blocks for which signature bitmaps have already
+   * been set.
```

Comment:
Ideally this functions interface could stay the same and we would figure out the intervals internal to this function. If the given validator was down for the whole window, the interval sizes don't really matter, so this function could provide reasonable defaults (e.g. 1000 blocks per interval)

Alternatively, this function could change and a new function could be created to provide a simple interface to execute a slash. Currently, it looks like any user interested in executing a slash needs to have a pretty intimate understanding of the internal details so they can post all the transactions they need to set the bitmaps before calling this function.

Note: I would not say this blocks the PR from being merged, but we will want to provide a nicer interface when possible so that making slashing bots is easy.

Reply:
I'm not sure I agree, this seems like the sort of off-chain logic that should be done in contractkit rather than internal to the contract

Reply:
Yes, that was something that we prioritised in the changes.
Use the contract as the simpler tool possible (with some limitations), but cover those limitations in the wrappers tools that we use.